lib/service/services_cli.rb

```diff
@@ -272,12 +272,36 @@ def install_service_file(service, file)
         odie "Formula `#{service.name}` has not implemented #plist, #service or installed a locatable service file"
       end
 
-      temp = Tempfile.new(service.service_name)
-      temp << if file.blank?
+      data = if file.blank?
         service.service_file.read
       else
         file.read
       end
+
+      if System.launchctl?
```

Comment:
I think this addition should happen when generating the file, not here.

Reply:
You mean `def generate_plist`? I tried that initially but it doesn't even seem to get called (at least for Apache). :>

Reply:
I mean in brew itself, in the `Service` class.

Reply:
It gets called when you install a formula, there's no real benefits in rewriting the plist at every start.

Reply:
But wouldn't that mean everyone who wants to use the new feature would have to reinstall **all** their services? I'm not entirely sure if it might even remove some config files.

Reply:
It would slowly propagate with software updates.

Reply:
Alright that works, plus reinstalling doesn't seem to remove configs so that should be fine too. I checked with redis because I noticed that your GitHub CI reported an error about it:
```
==>Successfully stopped `redis` (label: homebrew.mxcl.redis)
Bootstrap failed: 5: Input/output error
Error: Failure while executing; `/bin/launchctl bootstrap user/501 /usr/local/opt/redis/homebrew.mxcl.redis.plist` exited with 5.
Error: Process completed with exit code 1.
```

However, I can start/stop/restart without errors and even the exact commands being used in the test work fine:
```
brew services stop redis
sleep 5
brew services run redis
brew services restart redis
```

Anyways, the only thing this PR now changes is `gui/$UID` => `user/$UID`. :>